src/test/java/org/mabrarov/exceptionsafety/FactoryMethodTest.java

```diff
@@ -41,9 +43,31 @@
 
   private OutputStream createConfiguredResource()
       throws IOException, TestResourceConfigurationException {
-    final OutputStream resource = createResource();
-    configureResource(resource);
-    return resource;
+    OutputStream resource = createResource();
+    Throwable throwable = null;
+    Method method = new Object() {
+    }.getClass().getEnclosingMethod();
+    try {
+      configureResource(resource);
+      return resource;
+    } catch (final Throwable e) {
+      throwable = e;
+      for (Class exceptionClass : method.getExceptionTypes()) {
+        if (exceptionClass.isAssignableFrom(e.getClass())) {
+          //other exceptions handing
+          throw e;
```

Comment:
At the line 61 we do the same - just throw the same `e` exception. So what's the meaning of this `if` and `for`?